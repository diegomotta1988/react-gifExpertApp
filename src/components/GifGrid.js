import React from 'react';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from './../hooks/useFetchGifs';

import propTypes from 'prop-types';

export const GifGrid = ({ category }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category);
  // Al pasarle las dependencias como [] s√≥lo se ejecuta al renderizarse por primera vez el componente

  return (
    <>
      <h3 className='animate__animated animate__fadeIn'>{category}</h3>
      {loading && <p className='animate__animated animate__flash'>Loading</p>}
      <div className='card-grid'>
        {images.map((img) => (
          <GifGridItem {...img} key={img.id} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: propTypes.string.isRequired,
};
