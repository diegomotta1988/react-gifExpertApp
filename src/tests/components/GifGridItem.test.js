import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = 'Título test';
  const url = 'https://urldepruebas';
  const wrapper = shallow(<GifGridItem title={title} url={url} />);
  test('debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe tener un párrafo con el title', () => {
    const parrafo = wrapper.find('p');
    expect(parrafo.text().trim()).toBe(title);
  });

  test('debe tenre la imagen igual al url y alt de los props', () => {
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('debe tener animate__fadeIn ', () => {
    const div = wrapper.find('div');
    expect(div.hasClass('animate__fadeIn')).toBe(true);
  });
});
