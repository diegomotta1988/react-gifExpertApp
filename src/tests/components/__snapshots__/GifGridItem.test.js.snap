// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Título test"
    src="https://urldepruebas"
  />
  <p>
    Título test
  </p>
</div>
`;
